src/FastRoute/src/FastRouteKernel.cpp

```diff
@@ -361,18 +370,24 @@ void FastRouteKernel::runAntennaAvoidanceFlow()
     restorePreviousCapacities(_minRoutingLayer);
 
     _fastRoute->initAuxVar();
-    _fastRoute->run(newRoute);
-    addRemainingGuides(&newRoute);
+    _fastRoute->run(*newRoute);
+    addRemainingGuides(newRoute);
+    connectPadPins(newRoute);
     mergeResults(newRoute);
   }
+
+  for (FastRoute::NET& netRoute : *_result) {
+    mergeSegments(netRoute);
+  }
 }
 
 void FastRouteKernel::runClockNetsRouteFlow()
```

Comment:
again, should be a separate command, not more gook globed onto the existing command